cli/src/pager.rs

```diff
@@ -0,0 +1,135 @@
+use clap::{App, Arg, ArgMatches};
+use std::str::FromStr;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct Pager {
+    size: PageSize,
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum PageSize {
+    All,
+    Size(usize),
+}
+
+impl Pager {
+    pub fn add<'a, 'b>(app: App<'a, 'b>) -> App<'a, 'b> {
+        app.arg(
+            Arg::with_name("all")
+                .long("all")
+                .conflicts_with("page-size")
+                .help("Query for all records"),
+        ).arg(
+            Arg::with_name("page-size")
+                .long("page-by")
+                .conflicts_with("all")
+                .takes_value(true)
+                .validator(Self::validator)
+                .help("Query and display in a specific page size. Defaults to 10"),
+        )
+    }
+
+    fn validator(v: String) -> Result<(), String> {
+        usize::from_str(&v)
+            .map(|_| ())
+            .map_err(|_| String::from("Page size must be a positive integer"))
+    }
+
+    pub fn from_arg<'a>(arg: &'a ArgMatches) -> Pager {
+        if let Some(size) = arg.value_of("page-size") {
+            Pager {
+                size: PageSize::Size(usize::from_str(&size).unwrap_or(10)),
```

Comment:
There is the possibility that people pass large values in.  Is there an advantage to imposing a max on the size of `limit` vs validating that the value is < 200 on initialization?

Reply:
limit is the size of the request to horizon, I'm not worried about people asking for too much.

Reply:
I suppose i could rename the `limit` method to `horizon_page_limit`